pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -360,11 +377,17 @@ pub mod pallet {
 
 			Self::deposit_event(Event::PoolCreated { id: pool_id });
 
-			Ok(())
+			let currency_index = currency_length.saturated_into();
+			Ok(Some(match checked_curve {
```

Comment:
You return the same pre-computed weight, so this is not necessary.

Reply:
Oh yeah true: [a265987](https://github.com/KILTprotocol/kilt-node/pull/789/commits/a265987dbfef2f4b3ef9310f34625f7dbd9fc66a)

Reply:
Can you double check that there's no other place where you mistakenly do this?

Reply:
I checked the other extrinsics. Seems fine from my POV